File: 프로그래머스/0/120866. 안전지대/안전지대.swift
import Foundation

func solution(_ board:[[Int]]) -> Int {
    var result = board
    var points: [(Int, Int)] = []
    
    for y in result.indices {
        for x in result[y].indices {
            if result[y][x] == 1 { points.append((y, x))}
        }
    }
    
    points.forEach { y, x in
        for ny in y-1...y+1 {
            for nx in x-1...x+1 {
                guard 0 <= ny && ny < board.count else { continue }
                guard 0 <= nx && nx < board[0].count else { continue }
                guard (ny, nx) != (y, x) else { continue }
                guard board[ny][nx] == 0 else { continue }
                result[ny][nx] = 2
            }
        }
    }
    
    return result.flatMap { $0 }.filter{ $0 == 0 }.count
}

File: 프로그래머스/0/120866. 안전지대/안전지대.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

int solution(vector<vector<int>> board) {
    int answer = 0;
    vector<vector<int>> area(board.size(), vector<int>(board.size(), 0));
    int dy[8] = {-1, -1, 0, 1, 1, 1, 0, -1};
    int dx[8] = {0, 1, 1, 1, 0, -1, -1, -1};

    for (int i = 0; i < board.size(); i++) {
        for (int j = 0; j < board.size(); j++) {
            if (board[i][j]) {  // 지뢰가 매설된 지역이라면
                for (int k = 0; k < 8; k++) {
                    int ny = i + dy[k];
                    int nx = j + dx[k];
                    if (ny < 0 || ny >= board.size() || nx < 0 || nx >= board.size()) {
                        continue;
                    }
                    if (area[ny][nx]) {
                        continue;
                    }
                    area[ny][nx] = 1;
                }
                area[i][j] = 1;
            }
        }
    }

    for (int i = 0; i < board.size(); i++) {
        for (int j = 0; j < board.size(); j++) {
            if (area[i][j] == 0) {
                answer++;
            }
        }
    }

    return answer;
}